### PR TITLE
SPARK-8789 improve SQLQuerySuite resilience by dropping tables in setup

### PR DESCRIPTION
**Pull request by steveloughran**
drops tables used in tests before attempting to create them again. That way, if the previous test failed, the successors still work.

Alternative strategies would be to 
1. add try/finally clauses everywhere
2. have a teardown method which dropped listed tables, with test cases adding tables to the list to drop after creating them.

### CODE REVIEW

**Comment by cloud-fan** [comment]
We have a `SQLTestUtils` that can help us clean tables, maybe you can use that.

**Comment by steveloughran** [comment]
hmmm, its {{withTable}} would do it, wouldn't it?

**Comment by cloud-fan** [comment]
yea, you can make `SQLQuerySuite` extends `SQLTestUtils` and use `withTable` at every place that create table.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala [hunk 1]
```diff
@@ -62,6 +62,20 @@ class MyDialect extends DefaultParserDialect
  * valid, but Hive currently cannot execute it.
  */
 class SQLQuerySuite extends QueryTest {
+
+  /**
+   * Drops table `tableName` after calling `f`.
+   * This is the `SQLTestUtils` function without
+   * an explicit sql context
```
Comment (JoshRosen) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]: Why not just mix in `SQLTestUtils` then implement/override its `sqlContext` method to return `TestHive`?

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala [hunk 2]
```diff
@@ -603,37 +630,41 @@ class SQLQuerySuite extends QueryTest {
...
+      checkAnswer(
+        sql("SELECT key from explodeTest"),
+        (1 to 5).flatMap(i => Row(i) :: Row(i + 1) :: Nil)
+      )
 
-    sql("DROP TABLE explodeTest")
-    dropTempTable("data")
-    setConf(HiveContext.CONVERT_CTAS, originalConf)
+      dropTempTable("data")
+      setConf(HiveContext.CONVERT_CTAS, originalConf)
```
Comment (JoshRosen) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]: If you're going to change this code anyways, I guess that you might as well use `SQLTestUtils.withConf` here as well.

**Comment by marmbrus** [comment]
Hey @steveloughran, thanks for working on this!  Can we close this issue until you have time to bring it up to date and address @JoshRosen 's comments?

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala (hunk 1 above)
Comment (steveloughran) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]: there was a reason for that —I just forget what. Let me look at it & do it as a mixin

**Comment by steveloughran** [comment]
I'd forgotted about this issue & have been busy; I'll take it up again soon

**Comment by steveloughran** [comment]
Superceded by [https://github.com/apache/spark/pull/8748](https://github.com/apache/spark/pull/8748)

**Pull request closed without merging**